rlclientlib/schema/RankingEvent.fbs

```diff
@@ -11,7 +13,8 @@ table RankingEvent {
     probabilities:[float];    // probabilities
     model_id:string;          // model ID
     pass_probability:float;	  // Probability of event surviving throttling operation
-	meta:Metadata;
+    meta:Metadata;
+    decision_mode:DecisionMode; // decision mode used to determine rank behavior
```

Comment:
For reasons of backward/forward compatibility, 

1) use a uint/ubyte here.
2)  declare an enum for the value:

example: 
enum DecisionModeValues : ubyte { Online, Imitation }

This allows one to 
1) use enums in code 
2) enums are not enforced during deserialization so will not cause an error.
3) do proper min, max values checks in code and report error when deserialized value is outside min/max range

Reply:
Tyler suggested me adding the original array of actionId and probabilities in the ImitationMode structure. So We probably won't be able to use an enum.
 
From what I understand, the table is optional and struct is mandatory. We should be compatible.

I am working on this change.

Reply:
We had the discussion yesterday and decided to go with what you suggested here.

---
In reply to: [386537263](https://github.com/VowpalWabbit/reinforcement_learning/pull/129#discussion_r386537263) [](ancestors = 386537263)